### PR TITLE
polish codegen pai

### PR DESCRIPTION
**Pull request by typhoonzero**
Restructure PAI codegens

### CODE REVIEW

File: pkg/sql/codegen/pai/codegen_tf.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		table, err := formatODPSTables(resTable)
+		if err != nil {
+			return "", err
+		}
+		outputTables = fmt.Sprintf("-Doutputs=%s", table)
+	}
+	if cc.Worker.Count > 1 {
+		return fmt.Sprintf("pai -name tensorflow1120 -DjobName=%s -Dtags=dnn -Dscript=%s -DentryFile=entry.py -Dtables=%s %s -DcheckpointDir=\"%s\" -Dcluster=%s", jobName, tarball, submitTables, outputTables, ckpDir, cfQuote), nil
+	}
+	return fmt.Sprintf("pai -name tensorflow1120 -DjobName=%s -Dtags=dnn -Dscript=%s -DentryFile=entry.py -Dtables=%s %s -DcheckpointDir=\"%s\"", jobName, tarball, submitTables, outputTables, ckpDir), nil
```
Comment (weiguoz) [inline pkg/sql/codegen/pai/codegen_tf.go]: How about merging the two return statements into one? 
```go
cmd := fmt.Sprintf("pai -name tensorflow1150 -DjobName=%s -Dtags=dnn -Dscript=%s -DentryFile=entry.py -Dtables=%s %s -DcheckpointDir=\"%s\"", jobName, tarball, submitTables, outputTables, ckpDir)
if cc.Worker.Count > 1 {
    cmd = fmt.Sprintf("%s -Dcluster=%s", cmd, cfQuote)
}
return cmd, nil
```

File: pkg/sql/codegen/pai/cluster_config.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+// ClusterConfig implicates PAI distributed task meta
+type ClusterConfig struct {
+	PS     PSConfig     `json:"ps"`
+	Worker WorkerConfig `json:"worker"`
+}
+
+// GetClusterConfig returns ClusterConfig object comes from WITH clause
+func GetClusterConfig(attrs map[string]interface{}) (*ClusterConfig, error) {
```
Comment (wangkuiyi) [inline pkg/sql/codegen/pai/cluster_config.go:39]: From its content, the type ClusterConfig seems JobConfig?

File: pkg/sql/codegen/pai/codegen.go [hunk 3]
```diff
@@ -68,41 +45,6 @@ func FormatCkptDir(modelName string) (string, error) {
 	return strings.Join([]string{ossDir + "/", ossURIParts[1]}, "?"), nil
 }
```
Comment (wangkuiyi) [inline pkg/sql/codegen/pai/codegen.go:42]:
```go
func CheckpointURL(modelName string) (string, error) {
    dd := strings.Split(os.Getenv("SQLFLOW_OSS_CHECKPOINT_DIR"), "?")
    if len(dd) != 2 {
        return "", fmt.Errorf("Must set SQLFLOW_OSS_CHECKPOINT_DIR to oss://bucket/...?role_arn=...")
    } 
    return strings.Join([]string{path.Join(dd[0], modelName), dd[1]}, "?"), nil
}
```

File: pkg/sql/codegen/pai/codegen_random_forests.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
```
Comment (wangkuiyi) [inline pkg/sql/codegen/pai/codegen_random_forests.go]: `pkg/sql/codegen/pai/codegen_random_forests.go`  =>
`pkg/sql/codegen/pai/random_forest.go`

File: pkg/sql/codegen/pai/codegen.go [hunk 5]
```diff
@@ -68,41 +45,6 @@ func FormatCkptDir(modelName string) (string, error) {
...
-		},
-		Worker: WorkerConfig{
-			Count: defaultMap["train.num_workers"],
-			CPU:   defaultMap["train.worker_cpu"],
-			GPU:   defaultMap["train.worker_gpu"],
-		},
-	}, nil
-}
-
 func formatODPSTables(table string) (string, error) {
```
Comment (wangkuiyi) [inline pkg/sql/codegen/pai/codegen.go]: formatODPSTable => maxComputeTableURL

File: pkg/sql/codegen/pai/codegen_tf.go [hunk 6]
```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
```
Comment (wangkuiyi) [inline pkg/sql/codegen/pai/codegen_tf.go]: `pkg/sql/codegen/pai/codegen_tf.go` =>
`pkg/sql/codegen/pai/tensorflow.go`

File: pkg/sql/codegen/pai/cluster_config.go (hunk 2 above)
Comment (typhoonzero) [reply pkg/sql/codegen/pai/cluster_config.go:39]: Maybe the name `JobConfig` is a little bit generic, a job may also contains job name, job type etc.

File: pkg/sql/codegen/pai/codegen.go (hunk 3 above)
Comment (typhoonzero) [reply pkg/sql/codegen/pai/codegen.go:42]: Done. Not using `path.Join` because oss URIs must use `/` and it's not os specific.

**Review by weiguoz — APPROVED** [review]
LGTM.

**Pull request merged**